packages/nimble-components/src/combobox/specs/IxD.md

```diff
@@ -0,0 +1,212 @@
+
+#### Helper Text
+Only include helper text to provide information that aids the end-user in choosing the right selection. *See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information.*
+
+#### Field Input
+Comboboxes can be empty by default or have a default selection from the menu. Empty comboboxes should include placeholder text that propts the end-user to enter a custom value or select an option from the menu. Placeholder text should always follow the pattern "Enter or select [thing(s)]", for example "Enter or select address". Ellipses are not needed. Use sentence casing.
+
+![ ](spec-images/combobox-2.png)
+
+#### Menu & Menu Items
+A list of options to select, displayed as an open state.*See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information.*
+
+### Related Components
+To allow end-users to select a menu item but not enter a custom value, use a [**select**](/packages/nimble-components/src/select/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-select.png)
+
+To allow end-users to multi-select menu items, use a [**tag picker**](/packages/nimble-components/src/tag-picker/specs/IxD.md).
+
+![ ](spec-images/combobox-alternative-picker.png)
+
+## Design
+### Configuration
+#### Label
+The client-user should be able to specify where they want the label to be positioned. *See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information.*
+
+#### Icons
+The client-user should be able to specify whether they want left icons on menu items or not. *See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information.*
+
+#### Groups
+The client-user should be able to create groups for menu items. *See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information.*
+
+#### Secondary Text
+The client-user should be able to specify whether they want secondary text on menu items or not. *See [Select](/packages/nimble-components/src/select/specs/IxD.md) for more information.*
+
+#### Prefix & Suffix
+> NOTE: NEEDS VISUAL DESIGN
+
+The client-user should be able to specify a prefix and/or suffix that appears in the field.
+
+![ ](spec-images/combobox-3.png)
+
+#### Autocomplete
+The client-user should be able to specify autocomplete behavior (inline, list, both, none). Autocomplete is accessed via the field input (not pictured). *See [current implementation](https://nimble.ni.dev/storybook/?path=/docs/components-combobox--docs) for more information.*
+
+#### Add New
+> NOTE: NEEDS DEV REVIEW, VISUAL DESIGNER FOR STATES, & TECH WRITER
+
+The client-user should be able to specify whether custom entries can be added as a menu item in the menu or not. In the former, the menu should display a menu item that new items can be added indicate that a new item can be added. In the later, the menu does not show up at all.
```

Comment:
This feature I question the actual value of. In proper implementations within a UI framework like Angular, this addition of an item to the set of options in the dropdown will just happen automatically if a user commits a value that is currently not present in the set. This will happen in situations where the set of options are being generated from some observable, typically being populated by a service getting the set from some server state. It's not unreasonable to expect that when a user commits a new value that that value will makes its way to the server and ultimately update the observable that the `Combobox` is using to populate its options with.

In situations where the `Combobox` has some static set of options (so, not provided via an observable), then adding a new options to be used in the dropdown in successive operations isn't that useful, because it will ultimately be discarded on a page reload.

Ultimately, I'd like to hear the reasoning a client would have to want a feature like this, because at the moment I'm not sure what functionality it will truly provide.

Reply:
My thought process was that by default, the user can commit a value that isn't provided as an option in the menu, and while that value can be committed, it's not added as an option in the menu when that happens. I thought there is a use case for this when the end-user wants to assign a system (for example) to a workspace that doesn't exist yet, but that no longer appears to be accurate. 

I hadn't thought about it technically, but your points make sense. As I understand it, you are saying that the component itself is agnostic to whether the end-user can enter and commit a value that does not exist in the menu, because whether the commit is successful or errors is up to how it was implemented in an app by app developers. Is that correct? So there is no reason to have a specific "Add new" indicator in the dropdown, especially tied to some configuration on the component. But, like the loading indicator in the `select` spec, we may provide a slot in the menu and documented guidance about showing an "Add item" indicator.

Reply:
I think you are understanding things correctly. However, given that, I still don't think we would provide any documented guidance about an 'Add item' indicator. My concern there is that we would be possibly leading developers down an unnecessary road about how to introduce a new option into the option set, and we would have to provide pretty detailed instructions/caveats of what a user should expect if they were to introduce a mechanism like that into the menu.